lib/cadet_web/controllers/chat_controller.ex

```diff
@@ -0,0 +1,105 @@
+defmodule CadetWeb.ChatController do
+  use CadetWeb, :controller
+
+  use PhoenixSwagger
+
+  swagger_path :chat do
+    put("/chat")
+
+    summary("A wrapper for client that send queries to LLMs")
+
+    security([%{JWT: []}])
+
+    consumes("application/json")
+
+    parameters do
+      message(
+        :body,
+        :list,
+        "Conversation history. Need to be an non empty list of format {role: string, content:string}. For more details, refer to https://platform.openai.com/docs/api-reference/chat/create"
+      )
+    end
+
+    response(200, "OK")
+    response(400, "Missing or invalid parameter(s)")
+  end
+
+  def swagger_definitions do
+    %{
+      AdminSublanguage:
+        swagger_schema do
+          title("AdminSublanguage")
+
+          properties do
+            chapter(:integer, "Chapter number from 1 to 4",
+              required: true,
+              minimum: 1,
+              maximum: 4
```

Comment:
Although for now Chapters in SICP are only from 1-4, other numbers (even negative numbers) are valid combinations (e.g. Scheme, Python)

Reply:
Thanks for the comments.
This part is not relevant anymore and removed in the later commits